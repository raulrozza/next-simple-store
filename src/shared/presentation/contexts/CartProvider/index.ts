export { default as useAddToCart } from './hooks/useAddToCart';
