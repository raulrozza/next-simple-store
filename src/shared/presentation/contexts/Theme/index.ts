export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
