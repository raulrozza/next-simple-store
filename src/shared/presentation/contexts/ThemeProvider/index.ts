export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
