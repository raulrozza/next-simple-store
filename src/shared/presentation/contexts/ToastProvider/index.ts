export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
