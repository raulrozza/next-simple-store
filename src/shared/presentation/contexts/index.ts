export * from './CartProvider';
export * from './ThemeProvider';
