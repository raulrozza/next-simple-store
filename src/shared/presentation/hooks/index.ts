import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
