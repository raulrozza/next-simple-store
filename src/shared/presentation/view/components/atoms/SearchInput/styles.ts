import styled, { css } from 'styled-components';

export const StyledInput = styled.input`
    ${({ theme }) => css`
        background-color: ${theme.palette.gray['0']};
        border: 2px solid ${theme.palette.gray['100']};
        outline: none;
        border-radius: ${theme.layout.borderRadius.sm};

        width: 100%;
        padding: ${theme.layout.spacing(2)};
        transition: outline 0.2s, border 0.2s;

        &:focus {
            border-color: ${theme.palette.primary['400']};
        }
    `}
`;
