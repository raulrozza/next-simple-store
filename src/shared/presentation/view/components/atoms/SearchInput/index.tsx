import { FC, KeyboardEvent, useRef } from 'react';

import { StyledInput } from './styles';

interface SearchInputProps {
    onSubmit: (value: string) => void;
    placeholder: string;
}

const SearchInput: FC<SearchInputProps> = ({ onSubmit, placeholder }) => {
    const ref = useRef<HTMLInputElement>(null);

    const handleKeyUp = (event: KeyboardEvent<HTMLInputElement>) => {
        if (!ref.current) return;

        if (event.key === 'Enter') onSubmit(ref.current.value);
    };

    return (
        <StyledInput
            ref={ref}
            onKeyUp={handleKeyUp}
            placeholder={placeholder}
        />
    );
};

export default SearchInput;
