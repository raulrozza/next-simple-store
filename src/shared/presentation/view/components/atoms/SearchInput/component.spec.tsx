import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ThemeProvider } from '@/shared/presentation/contexts';

import SearchInput from './index';

describe('SearchInput', () => {
    it('should render the input correctly', () => {
        const view = render(
            <SearchInput placeholder="My input" onSubmit={jest.fn()} />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByPlaceholderText('My input')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('My input')).toHaveValue('');
        expect(view).toMatchSnapshot();
    });

    it('should change the field value when typing the input', async () => {
        render(<SearchInput placeholder="My input" onSubmit={jest.fn()} />, {
            wrapper: ThemeProvider,
        });

        const input = screen.getByPlaceholderText(/My input/);
        await userEvent.type(input, 'new value');

        expect(screen.getByPlaceholderText(/My input/)).toHaveValue(
            'new value',
        );
    });

    it('should call the on submit function when typing enter', async () => {
        const onSubmit = jest.fn();

        render(<SearchInput placeholder="My input" onSubmit={onSubmit} />, {
            wrapper: ThemeProvider,
        });

        const input = screen.getByPlaceholderText(/My input/);
        await userEvent.type(input, 'new value{Enter}');

        expect(onSubmit).toHaveBeenCalledWith('new value');
    });
});
