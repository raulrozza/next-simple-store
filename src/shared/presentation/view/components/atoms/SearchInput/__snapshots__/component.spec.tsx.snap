// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render the input correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c0:focus {
  border-color: #8CCFDE;
}

<body>
    <div>
      <input
        class="c0"
        placeholder="My input"
      />
    </div>
  </body>,
  "container": .c0 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c0:focus {
  border-color: #8CCFDE;
}

<div>
    <input
      class="c0"
      placeholder="My input"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
